scripts/apps/search/components/MarkedDesksList.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import {
+    closeActionsMenu
+} from 'apps/search/helpers';
+
+/**
+ * @ngdoc React
+ * @module superdesk.apps.search
+ * @name MarkedDesksList
+ * @param {Object} svc the services nedded: desks and $timeout
+ * @param {Object} item story to be marked
+ * @param {Object} markedDesksById the dict of deskId and desk object
+ * @description Creates a list of desks that is used for marking a story for a desk
+ */
+export class MarkedDesksList extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.removeMarkedDesk = this.removeMarkedDesk.bind(this);
+        this.stopTimeout = this.stopTimeout.bind(this);
+        this.close = this.close.bind(this);
+    }
+
+    removeMarkedDesk(desk) {
+        const {desks} = this.props.svc;
+
+        return function(event) {
+            event.stopPropagation();
+            desks.markItem(desk._id, this.props.item);
+        }.bind(this);
+    }
+
+    componentDidMount() {
+        this.timeout = null;
+    }
+
+    componentWillUnmount() {
+        this.stopTimeout();
+    }
+
+    stopTimeout() {
+        const {$timeout} = this.props.svc;
+
+        this.timeout = $timeout.cancel(this.timeout);
+    }
+
+    close() {
+        const {$timeout} = this.props.svc;
+
+        this.timeout = $timeout(closeActionsMenu, 2000, false);
+    }
+
+    render() {
+        const {desks} = this.props.svc;
+        var markedDesks = this.props.item.marked_desks;
+        var markedDesksById = this.props.markedDesksById || {};
+
+        return (
+            <ul className="dropdown dropdown__menu highlights-list-menu open"
+            onMouseEnter={this.stopTimeout} onMouseLeave={this.close}>
+                <li key="item-marked-label">
+                    <div className="dropdown__menu-label">{gettext('Marked For')}
+                        <button className="dropdown__menu-close" onClick={closeActionsMenu}>
+                            <i className="icon-close-small icon-white" />
+                        </button>
+                    </div>
+                </li>
+                {
+                    markedDesks.map((d) => <li key={'item-marked-' + markedDesksById[d.desk_id]._id}>
+                    {markedDesksById[d.desk_id].name}
+                    {desks.hasMarkItemPrivilege() ? <button className="btn btn--mini"
+                        onClick={this.removeMarkedDesk(markedDesksById[d.desk_id])}>
+                            {gettext('REMOVE')}</button> : null}
+                    </li>)
+            }
+                </ul>
+        );
+    }
+}
+
+/**
+ * svc: contains $timeout and desks and is required
```

Comment:
It's enough that you've documented these in the `params` section of the component. Sorry if I didn't notice this before. This doesn't hurt though! Thanks for adding!